### DISCUSSION TITLE
how to use local-session in url

### DISCUSSION THREAD

**Discussion by Jasonzzzzzzzzzzzzzz** [GMOD/jbrowse-components#4267] [Q&A] [answered]
I save a session from GUI, then how can I use the session? Where should I put the session in(config.json or other place)? I have try put the session in my config.json, and my url query look like this "?config=config.json&session=local-xxxxxx", but it can't work.

**Comment by cmdcolin** [comment]
@Jasonzzzzzzzzzzzzzz hi there
generally you would use the "share session" button

screenshot:
![Screenshot from 2024-03-07 13-36-42](https://github.com/GMOD/jbrowse-components/assets/6511937/39e0e5b8-fa85-4b95-8f07-fd48f0c5f9fe)


this would allow re-sharing the result to other users

the "local" session URLs are not shareable
this doc link covers the different type of session links briefly
https://jbrowse.org/jb2/docs/urlparams/#other-session-options

note that it is not possible to share "jbrowse desktop" with web currently (not sure if that was what you are asking, but thought i'd mention it)

**Reply by cmdcolin** [reply]
also, I forgot to mention, if you do want to encode the session into your config.json like you mention, you can copy it into your "defaultSession" variable. basically the session from "File->Export session" will be a JSON file with a key named "session". just copy that key's value into a variable named defaultSession in a config.json

**Reply by Jasonzzzzzzzzzzzzzz** [reply]
Really thanks for your reply. May my question is not very clearly. I deployed jbrowser2 in my server, I want user can see these things when they open the website (some tracks are selected and the hierarchicalTrackSelector is opened).
![image](https://github.com/GMOD/jbrowse-components/assets/49664139/86134ef7-083d-445a-8f91-ea4e009a9582)
I know one way is use share session in my url, like https://myip/?session=share-xxxxx.
But I notice the first demo in Jbrowser2, the url is https://jbrowse.org/code/jb2/v2.10.3/?config=test_data%2Fconfig_demo.json&session=local-WxzwtvYLzblmkxu18p15P
In this url , it has a local session(session=local-WxzwtvYLzblmkxu18p15P). So my question is how can I make local session like this demo?
Thanks~

**Reply by cmdcolin** [reply, marked as answer]
" I want user can see these things when they open the website (some tracks are selected and the hierarchicalTrackSelector is opened)."

you might consider using the URL param api, it has an easier to use interface than sessions

example

 https://myip/?loc=1:1-100&assembly=Mmul_10&tracklist=true&tracks=track1,track2,track3

this navigates to a particular location, for a particular assembly, with some tracks turned on by default, and with the track selector turned on by default, on page load


you can put this URL as the iframe src if you are using iframe embedding

currently the "sessions" are difficult to construct programmatically, so the &tracks, &loc, &assembly, etc are a simplified method for making sessions programmatically

see here for the docs for that url param api https://jbrowse.org/jb2/docs/urlparams/

**Reply by Jasonzzzzzzzzzzzzzz** [reply]
Got it! Really Thanks!
